source/faucet/main.d

```diff
@@ -56,26 +56,90 @@ import vibe.web.rest;
 
 static immutable KeyCount = WK.Keys.byRange().length;
 
-/// Configuration parameter for Faucet
-private struct Config
+/// The keys that will be used for generating transactions
+private SecretKey[PublicKey] secret_keys;
+
+/// The configuration for faucet as a faucet and a tx generator
+private Config config;
+
+private struct TxGenerator
 {
+   /// Stats port (default: 9113)
+    ushort stats_port;
+
     /// How frequently we run our periodic task
-    static immutable interval = 30.seconds;
+    ulong send_interval;
 
     /// Between how many addresses we split a transaction by
-    static immutable count = 15;
+    uint split_count;
 
-    /// Bind address
-    public string address;
+    /// Maximum number of utxo before merging instead of splitting
+    uint merge_threshold;
 
-    /// Bind port
-    public ushort port;
-
-    /// Stats port (default: 9113)
-    public ushort stats_port = 9113;
+    /// Addresses to send the transactions to
+    string[] addresses;
 
     /// Keys from the config
-    SecretKey[PublicKey] keys;
+    string[] keys;
+
+    this (ulong send_interval, uint split_count, uint merge_threshold,
+        string[] addresses, string[] keys)
+    {
+        this.stats_port = stats_port;
+        this.send_interval = send_interval;
+        this.split_count = split_count;
+        this.merge_threshold = merge_threshold;
+        this.addresses = addresses;
+        this.keys = keys;
+    }
+
+    this (Node node) @safe
+    {
+        stats_port = node["stats_port"].as!ushort;
+        send_interval = node["send_interval"].as!ulong;
+        split_count = node["split_count"].as!uint;
+        merge_threshold = node["merge_threshold"].as!uint;
+        () @trusted { addresses = parseSequence("addresses", node, true); }();
+        () @trusted { keys = parseSequence("keys", node, false); }();
+    }
+}
+
+/// The Config for the faucet web
+private struct Web
+{
+    /// Address to bind for website
+    string address;
+
+    /// Port to bind for website
+    ushort port;
+
+    this (string address, ushort port = 9113)
```

Comment:
This port default should not be 9113 as it clashes with stats default port. I'll update it.

Reply:
Updated

Reply:
Good catch..